collector/netstats/netstats.go

```diff
@@ -77,6 +79,12 @@ type Interface interface {
 
 	// CountSend reports inbound bytes.
 	CountReceive(ctx context.Context, ss SizesStruct)
+
+	// SetSpanAttributes takes a context and adds attributes to the associated span.
+	SetSpanSizeAttributes(ctx context.Context, ss SizesStruct)
+
+	// SetSpanError set span status explicitly, if there is an non-nil error provided.
+	SetSpanError(ctx context.Context, err error)
```

Comment:
I would prefer to see the two lines of code underneath this interface directly written at the call sites. The method doesn't really relate to the netstats module, e.g., note that the body of
```
func (rep *NetworkReporter) SetSpanError(ctx context.Context, err error)
```
does not refer to `rep`.

Reply:
Thanks for the review! Makes sense removed this from the netstats package